### PR TITLE
Implement Save and Discard Mechanism for Changes in Payrun

### PR DESCRIPTION
**Pull request by Publicker**
This PR introduces a new feature in the Payrun "configuration" process, including control over changes made to the selected payment account per currency and the inclusion of invoices in a payrun.

# Changes
- **New Save and Discard Buttons**: When changes are made to the account selected for payment or to the status of invoices being included or excluded from the payrun, "Save" and "Discard" buttons now appear. These buttons allow users to either commit their changes or revert them to the last saved state.
- **Confirmation on Navigation**: If a user attempts to navigate away from the page by clicking on "All Payruns" while unsaved changes are present, a native popup will warn them that leaving the page will result in the loss of those changes. This provides an opportunity to cancel the navigation, save the changes, and then proceed as intended.

# Big Notes/Caveats
- **Local Saving of Changes**: Currently, the changes are only saved locally. This limitation is due to the pending API work required to persist these changes server-side. As a result, if the page is reloaded, any unsaved changes will be lost.
- **Temporary UI Design**: The design of the "Save" and "Discard" buttons is temporary and subject to change. The current UI was implemented to allow for the development and testing of functionality. When design is ready, it will be addressed.

# Demo Video
https://github.com/nofrixion/nofrixion.business/assets/52673485/0e515637-8ead-44af-87bb-10faf3be3b33

### CODE REVIEW

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/192.

File: packages/clients/src/types/ApiResponses.ts [hunk 1]
```diff
@@ -588,6 +588,7 @@ export interface Payrun {
   inserted: Date
   lastUpdated: Date
   scheduleDate?: Date
+  accountsToPay?: Record<Currency, string>
```
Comment (sauravmaiti22) [inline packages/clients/src/types/ApiResponses.ts:591]: I believe this is the source accounts to pay from. accountsToPay sounds more like the recipient account. Perhaps, we can call it sourceAccounts or accountsToPayFrom?

Comment (donalnofrixion) [reply packages/clients/src/types/ApiResponses.ts:591]: +1 for sourceAccounts

**Review by donalnofrixion — CHANGES_REQUESTED** [review]

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/192.

**Review by donalnofrixion — APPROVED** [review]

**Review by sauravmaiti22 — APPROVED** [review]

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
Thanks Pablo.

- I think the buttons work well so far. If we encounter any issues in the future we can think of making special buttons.
- The button should say "Save changes".
- The width of the save button should adapt to the text length.
- If the user makes a change and then changes it again to go back to the last saved version, the save options should go away.


- Let's use this dialog:

> image

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/192.

**Comment by Publicker** [comment]
Thanks, @pabloi-nf. Comments have been addressed.

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
Thanks!
The leave dialog needs to appear on any link that takes the user out of this screen.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/192.

**Comment by Publicker** [comment]
Should behave like that now @pabloi-nf

**Comment by Publicker** [comment]
This part of my life … this little part … is called … Happiness gif

**Pull request merged**